crates/runtime/src/lib.rs

```diff
@@ -283,3 +283,8 @@ impl fmt::Display for WasmFault {
         )
     }
 }
+
+#[cfg(all(target_arch = "s390x", feature = "versioned-exports"))]
+compile_error!(
+    "Feature `versioned-exports` is not currently supported for the `s390x` architecture"
+);
```

Comment:
One idea for handling this is to use preprocessor magic on s390x which supports I think `foo ## bar` to create the symbol `foobar` so Wasmtime could invoke the C compiler with something like `-DSYMBOL_APPEND=10_0_0` or something like that.